executor/common_bsd.h

```diff
@@ -153,6 +167,28 @@ static void initialize_tun(int tun_id)
 	char local_ipv6[sizeof(LOCAL_IPV6)];
 	snprintf_check(local_ipv6, sizeof(local_ipv6), LOCAL_IPV6, tun_id);
 	execute_command(1, "ifconfig %s inet6 %s", tun_iface, local_ipv6);
+
+#if GOOS_openbsd
+	// Connect tun and bpf devices. Since /dev/bpf is a cloning device on
+	// OpenBSD, we don't have to bother opening a unique device per executor
+	// process.
+	bpffd = open("/dev/bpf", O_RDWR | O_NONBLOCK);
+	if (bpffd < 0)
+		fail("bpf: can't open: /dev/bpf");
+
+	// Remap bpf onto higher fd number to hide it from fuzzer and to keep
+	// fd numbers stable regardless of whether bpf is opened or not (also see kMaxFd).
+	const int kBpfFd = 239;
+	if (dup2(bpffd, kBpfFd) < 0)
+		fail("dup2(bpffd, kBpfFd) failed");
+	close(bpffd);
+	bpffd = kBpfFd;
+
+	struct ifreq ifr;
+	strlcpy(ifr.ifr_name, tun_iface, sizeof(ifr.ifr_name));
+	if (ioctl(bpffd, BIOCSETIF, &ifr) < 0)
```

Comment:
I think it's better to keep this in declarative descriptions. It seems that this can be reduces to something like:
```
ioctl$BIOCSETIF(fd fd_bpf, cmd const[BIOCSETIF], arg ptr[in, ifreq_name])

ifreq_name {
	ifr_name	stringnoz["tap"]
	tapindex	proc['0', 1, int8]
	z               const[0, int8]
} [size[IFREQ_SIZE]]

define IFREQ_SIZE sizeof(struct ifreq)
```
Later we can add more interface names, more ioctls that operate on devices, it will allow it to create several bpf's and arbitrary re-attach them to interfaces, etc.
We use the C code when descriptions fail miserably and custom code only points to areas that descriptions can't describe.

Reply:
Nice, didn't know. Updated PR.

Reply:
Just did some light testing and the `ioctl$BIOCSETIF` description works.